### PR TITLE
Add support for php 8 & update dependencies to their latest versions

### PR DESCRIPTION
**Pull request by TomKeyte**
PHP 8 has been released.

This PR adds support for it, as well as updates the dependencies to their latest versions

### CODE REVIEW

**Comment by bartfeenstra** [comment]
This looks good! To confirm this all works on PHP 8, can you update `.travis.yml`?

**Comment by TomKeyte** [comment]
> This looks good! To confirm this all works on PHP 8, can you update `.travis.yml`?

Will do.

Perhaps I should remove support for PHP 5, and <=7.3 since they are all EOL?

**Comment by bartfeenstra** [comment]
Removing support for EOL versions would be a BC break and require a new major version release. First of all that would therefore have to be done in a separate PR, but as I am no longer actively maintaining this package that seems like an unnecessary disruption at this point.

**Comment by TomKeyte** [comment]
Apologies, forgot to get around to this

Haven't used travis before - hoping just adding php 8 will be enough :)

File: .phpunit.result.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...ommercie\Tests\Currency\InputTest::testParseAmount";d:0;s:67:"Commercie\Tests\Currency\ResourceRepositoryTest::testListCurrencies";d:0.001;s:65:"Commercie\Tests\Currency\ResourceRepositoryTest::testLoadCurrency";d:0.004;s:107:"Commercie\Tests\Currency\ResourceRepositoryTest::testLoadCurrencyWithNonExistentCurrencies with data set #0";d:0;s:107:"Commercie\Tests\Currency\ResourceRepositoryTest::testLoadCurrencyWithNonExistentCurrencies with data set #1";d:0;s:107:"Commercie\Tests\Currency\ResourceRepositoryTest::testLoadCurrencyWithNonExistentCurrencies with data set #2";d:0;s:107:"Commercie\Tests\Currency\ResourceRepositoryTest::testLoadCurrencyWithNonExistentCurrencies with data set #3";d:0;s:107:"Commercie\Tests\Currency\ResourceRepositoryTest::testLoadCurrencyWithNonExistentCurrencies with data set #4";d:0;s:107:"Commercie\Tests\Currency\ResourceRepositoryTest::testLoadCurrencyWithNonExistentCurrencies with data set #5";d:0;s:57:"Commercie\Tests\Currency\Resources::testCurrencyIntegrity";d:0.029;s:48:"Commercie\Tests\Currency\UsageTest::testGetStart";d:0;s:46:"Commercie\Tests\Currency\UsageTest::testGetEnd";d:0;s:54:"Commercie\Tests\Currency\UsageTest::testGetCountryCode";d:0;}}}
```
Comment (bartfeenstra) [inline .phpunit.result.cache]: This file must be removed. It was probably added before `.gitignore` was updated :)

**Review by bartfeenstra — CHANGES_REQUESTED** [review]

**Comment by TomKeyte** [comment]
Updated

**Review by bartfeenstra — APPROVED** [review]

**Comment by bartfeenstra** [comment]
Thanks! It seems Travis has made some changes in recent months and I'm working on re-enabling builds so we can confirm everything works on PHP 8.

**Comment by bartfeenstra** [comment]
I can't see why this repository is now showing up in my Travis CI dashboard, and without that happening I cannot reenable builds to ensure these changes work. As I have moved away from e-commerce some years ago and no longer maintain this or any other e-commerce package, as well as no longer use Travis CI for any of my active repositories, unfortunately I have limited time to spend on this.

**Comment by Berdir** [comment]
Merging this, I did run the tests manually on PHP 8, created a follow-up to convert the travis stuff to github actions: #7

**Comment by Berdir** [comment]
One note:

> Removing support for EOL versions would be a BC break and require a new major version release.

Note that semver allows to change versions and even add/remove dependencies in minor versions, and strictly speaking, the changes here require 7.1 at least (the void return type), so it would have been correct update that accordingly, but it only affects running tests. 

I did make that change in https://github.com/Commercie/currency-exchange/pull/3 though. might update this here if/when working on #7.

**Pull request merged**